shared/naturalcrit/splitPane/splitPane.jsx

```diff
@@ -1,200 +1,121 @@
-					<i className='fas fa-circle' />
-					<i className='fas fa-circle' />
+		const newSize = limitPosition(e.pageX);
+		setDividerPos(newSize);
+	};
+
+	const liveScrollToggle = ()=>{
+		window.localStorage.setItem('liveScroll', String(!liveScroll));
+		setLiveScroll(!liveScroll);
+	};
+
+	const moveArrows = showMoveArrows && (
+		<>
+			{['left', 'right'].map((direction, index) => (
+				<div
+					key={direction}
+					className={`arrow ${direction}`}
+					onClick={index === 0 ? () => setMoveSource(!moveSource) : () => setMoveBrew(!moveBrew)}
+				>
+					<i className={`fas fa-arrow-${direction}`} />
 				</div>
+			))}
+			<div
+				id='scrollToggleDiv'
+				className={`arrow ${liveScroll ? 'lock' : 'unlock'}`}
+				onClick={liveScrollToggle}
+			>
+				<i id='scrollToggle' className={`fas fa-${liveScroll ? 'lock' : 'unlock'}`} />
+			</div>
+		</>
+	);
+
+	const renderDivider = ()=>(
+		<div className='divider' onPointerDown={handleDown} ref={dividerRef}>
+			{showDividerButtons && moveArrows}
+			<div className='dots'>
+				<i className='fas fa-circle' />
+				<i className='fas fa-circle' />
+				<i className='fas fa-circle' />
 			</div>
-		</>;
-	},
+		</div>
+	);
 
-	render : function(){
-		return <div className='splitPane' onPointerMove={this.handleMove} onPointerUp={this.handleUp}>
-			<Pane
-				width={this.state.currentDividerPos}
-			>
-				{React.cloneElement(this.props.children[0], {
-					...(this.props.showDividerButtons && {
-						moveBrew      : this.state.moveBrew,
-						moveSource    : this.state.moveSource,
-						liveScroll    : this.state.liveScroll,
-						setMoveArrows : this.setMoveArrows,
-					}),
-				})}
+	return (
+		<div className='splitPane' onPointerMove={handleMove} onPointerUp={handleUp}>
+			<Pane width={dividerPos} moveBrew={moveBrew} moveSource={moveSource} liveScroll={liveScroll} setMoveArrows={setShowMoveArrows}>
+				{props.children[0]}
 			</Pane>
-			{this.renderDivider()}
-			<Pane isDragging={this.state.isDragging}>{this.props.children[1]}</Pane>
-		</div>;
-	}
-});
-
-const Pane = createClass({
-	displayName     : 'Pane',
-	getDefaultProps : function() {
-		return {
-			width : null
-		};
-	},
-	render : function(){
-		let styles = {};
-		if(this.props.width){
-			styles = {
-				flex  : 'none',
-				width : `${this.props.width}px`
-			};
-		} else {
-			styles = {
-				pointerEvents : this.props.isDragging ? 'none' : 'auto' //Disable mouse capture in the rightmost pane; dragging into the iframe drops the divider otherwise
-			};
-		}
-
-		return <div className={cx('pane', this.props.className)} style={styles}>
-			{this.props.children}
-		</div>;
-	}
-});
+			{renderDivider()}
+			<Pane isDragging={isDragging}>{props.children[1]}</Pane>
+		</div>
+	);
+};
+
+const Pane = ({ width, children, isDragging, className, moveBrew, moveSource, liveScroll, setMoveArrows })=>{
+	const styles = width
+		? { flex: 'none', width: `${width}px` }
+		: { pointerEvents: isDragging ? 'none' : 'auto' };
```

Comment:
Let's keep this comment

```suggestion
		: { pointerEvents: isDragging ? 'none' : 'auto' }; //Disable mouse capture in the right pane; else dragging into the iframe drops the divider
```